### PR TITLE
Remove collection and role path adulterating

### PR DESCRIPTION
**Pull request by ssbarnea**
This major change removes all the core related to modification of ansible include paths for collections and roles.

This also replaces the dependency on ansible-lint with ansible-compat thin library, one that aims to provide exactly the bootstrapping functionality both projects need.

Once this is merged, molecule itself will have no special knowledge of these folders and will make its behavior less likely to be different than just Ansible execution.

Molecule will rely on ansible-compat runtime functionality for injecting temporary roles and collections include paths and installing the requirements.

Who is unlikely to impacted:
* you have `requirements.yml` at repository root or scenario folder
* you used shell tasks running `ansible-galaxy role|collection install`, but did not specify a custom target path

Who is likely to be impacted:
* You manually modified ansible collections paths or ansible roles paths inside molecule.yml file
* You used custom path locations when installing roles or collections

Related: #3171

### CODE REVIEW

**Comment by ssbarnea** [comment]
@tadeboro @felixfontein Do you think that we should remove any code in molecule that modifies the collection and role include paths?

I am asking this because I think we no longer need it as the prerun functionality can take care of these. The reported bugs with dependency are related to conflict between what molecule did and the newer cache.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -50,11 +50,11 @@ repos:
     hooks:
       - id: mypy
         # empty args needed in order to match mypy cli behavior
-        args: []
+        args: ['--no-incremental']
         entry: mypy src/
         pass_filenames: false
         additional_dependencies:
-          - ansible-lint>=5.0.10
+          - ansible-compat>=0.2.0
```
Comment (felixfontein) [inline .pre-commit-config.yaml:57]: I don't think we should depend on non-stable releases, and once there is a stable release, we should use semver to make sure there are no breaking changes.
```suggestion
          - ansible-compat>=1.0.0,<2.0.0
```

File: setup.cfg [hunk 2]
```diff
@@ -66,7 +66,7 @@ setup_requires =
 
 # These are required in actual runtime:
 install_requires =
-    ansible-lint >= 5.0.12  # only for the prerun functionality
+    ansible-compat >= 0.2.0
```
Comment (felixfontein) [inline setup.cfg:69]: Same as above
```suggestion
    ansible-compat >= 1.0.0, < 2.0.0
```

**Comment by felixfontein** [comment]
I'm not using molecule so I can't really say much about this. I personally prefer less magic, so the idea sounds good to me.

**Pull request closed without merging**